### PR TITLE
Add shfmt.

### LINKED ISSUE #94: Add support for shell/bash via shfmt
Which can be installed via `go get -u mvdan.cc/sh/cmd/shfmt`.

The tool works in a very similar way to `gofmt`, so I figure that it would be simple to add support for it. Happy to help with any issues that may arise with it.

### PR DESCRIPTION
**Pull request by infokiller**
Fixes #94

### CODE REVIEW

**Comment by infokiller** [comment]
I signed it!

File: autoload/codefmt/shfmt.vim [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+let s:plugin = maktaba#plugin#Get('codefmt')
+
+
+""
+" @private
+" Formatter: shfmt
+function! codefmt#shfmt#GetFormatter() abort
+  let l:formatter = {
+      \ 'name': 'shfmt',
+      \ 'setup_instructions': 'Install shfmt and configure the ' .
```
Comment (dbarnett) [inline autoload/codefmt/shfmt.vim]: This should link to where you can get shfmt or find out more about it. There's nothing else to disambiguate if another tool called shfmt comes along, whereas the code is crafted to work with that particular tool.

File: vroom/shfmt.vroom [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+shfmt_executable flag if the default of "shfmt" doesn't work.
+
+  :Glaive codefmt shfmt_executable='myshfmt'
+  :FormatCode shfmt
+  ! myshfmt .*
+  $ f()
+  :Glaive codefmt shfmt_executable='shfmt'
+
+
+You can format any buffer with shfmt specifying the formatter explicitly.
```
Comment (dbarnett) [inline vroom/shfmt.vroom]: Could you also include a realistic case for failure handling to demonstrate the quickfix stuff is working?

File: autoload/codefmt/shfmt.vim [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    if type(l:Shfmt_options) is# type([])
+      let l:shfmt_options = l:Shfmt_options
+    elseif maktaba#value#IsCallable(l:Shfmt_options)
+      let l:shfmt_options = maktaba#function#Call(l:Shfmt_options)
+    else
+      throw maktaba#error#WrongType(
+          \ 'shfmt_options flag must be list or callable. Found %s',
+          \ string(l:Shfmt_options))
+    endif
+    " Hack range formatting by formatting range individually, ignoring context.
```
Comment (dbarnett) [inline autoload/codefmt/shfmt.vim]: Could you follow up with the tool maintainer to see about adding range formatting?

**Comment by infokiller** [comment]
Done.

File: vroom/shfmt.vroom (hunk 2 above)
Comment (infokiller) [reply vroom/shfmt.vroom]: Done.

**Comment by infokiller** [comment]
Fixed the test

**Comment by dbarnett** [comment]
Thanks! Merged.

**Pull request merged**